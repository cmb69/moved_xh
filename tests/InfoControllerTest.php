<?php

/**
 * Copyright 2023 Christoph M. Becker
 *
 * This file is part of Moved_XH.
 *
 * Moved_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Moved_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Moved_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Moved;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;

class InfoControllerTest extends TestCase
{
    public function testDefaultActionRendersPluginInfo(): void
    {
        $plugin_tx = XH_includeVar("./languages/en.php", 'plugin_tx');
        $lang = $plugin_tx['moved'];
        $systemChecker = $this->createStub(SystemChecker::class);
        $systemChecker->method('checkVersion')->willReturn(true);
        $systemChecker->method('checkWritability')->willReturn(true);
        $sut = new InfoController("./", $lang, $systemChecker);
        $response = $sut->defaultAction();
        Approvals::verifyHtml($response);
    }
}
