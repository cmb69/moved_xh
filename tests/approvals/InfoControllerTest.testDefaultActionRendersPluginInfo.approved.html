
<h1>Moved 1.0beta4</h1>
<div>
    <h2>System check</h2>
    <p class="xh_success">Checking that the PHP version is at least 7.1.0 … okay</p>
    <p class="xh_success">Checking that the CMSimple_XH version is at least 1.7.0 … okay</p>
    <p class="xh_success">Checking that './css/' is writable … okay</p>
    <p class="xh_success">Checking that './languages/' is writable … okay</p>
</div>
